Lib/test/test_urllib2.py

```diff
@@ -173,12 +172,29 @@ def test_password_manager(self):
 
         add("c", "http://example.com/foo", "foo", "ni")
         add("c", "http://example.com/bar", "bar", "nini")
+        add("c", "http://example.com/foo/bar", "foobar", "nibar")
```

Comment:
This line

> add("c", "http://example.com/foo/bar", "foobar", "nibar")

along with this test a line 184 confuses me.

>         self.assertEqual(find_user_pass("c", "http://example.com/foo/bar"), ('foo', 'ni'))

1. If we wanted to highlight that only `/foo` path's password will be used. A comment above line 175 will be helpful.
2. The behavior is a bit confusing too. Cannot `/foo/bar` have it's own username:password?

Reply:
This behavior seems intentional. See a comment at line 155: "For the same realm, password set the highest path is the winner." Tests at lines 158, 164 and 167 ensure this, but only for the root path. I added new tests for non-root path.

Should I repeat this comment above line 175?

Reply:
Oh I see. I missed seeing the context in the diff. I notice an example illustrating the behavior for the comment already present in line 149 and line 150. 

So, I don't think we need another comment. Thank you.